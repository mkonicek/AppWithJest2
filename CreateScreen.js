import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

export default class CreateScreen extends React.Component {

  constructor(props) {
   super(props);
   this.state = {isHelpExpanded: false};
  }
  onHelpPress = () => {
    this.setState({isHelpExpanded: !this.state.isHelpExpanded});
  }
  render() {
    return (
      <View>
        <TouchableOpacity onPress={this._onHelpPress}>
          <Text>See Help</Text>
        </TouchableOpacity>
        {this.state.isHelpExpanded &&
        <View>
          <Text>This is some help text</Text>
        </View>}
      </View>
    );
  }
}
