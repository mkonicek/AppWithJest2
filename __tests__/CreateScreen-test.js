import 'react-native';
import React from 'react';
import CreateScreen from '../CreateScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// Advanced use case - simulate button tap
describe('CreateScreen', () => {

  it('expands help tooltip when tapped', () => {
    const component = renderer.create(
      <CreateScreen />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.getInstance().onHelpPress();
    // Simulate button click - how?
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
