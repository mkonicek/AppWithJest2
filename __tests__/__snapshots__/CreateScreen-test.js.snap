exports[`CreateScreen expands help tooltip when tapped 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text>
      See Help
    </Text>
  </View>
</View>
`;

exports[`CreateScreen expands help tooltip when tapped 2`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text>
      See Help
    </Text>
  </View>
  <View>
    <Text>
      This is some help text
    </Text>
  </View>
</View>
`;
